src/js/game.js

```diff
@@ -0,0 +1,185 @@
+import { $, addEvent } from './utils.js';
+import { MESSAGE } from './constants.js';
+import Car from './car.js';
+
+export default class RacingGame {
+  constructor(container) {
+    this.container = container;
+    this.cars = [];
+    this.gameCount = 0;
+
+    this.init();
+    this.bindEvent();
+  }
+
+  init() {
+    this.carnameContainer = $({
+      target: this.container,
+      selector: '.carname-container',
+    });
+    this.carnameInput = $({
+      target: this.carnameContainer,
+      selector: 'input',
+    });
+    this.carnameButton = $({
+      target: this.carnameContainer,
+      selector: 'button',
+    });
+
+    this.gamecountContainer = $({
+      target: this.container,
+      selector: '.gamecount-container',
+    });
+    this.gamecountInput = $({
+      target: this.gamecountContainer,
+      selector: 'input',
+    });
+    this.gamecountButton = $({
+      target: this.gamecountContainer,
+      selector: 'button',
+    });
+
+    this.trackContainer = $({
+      target: this.container,
+      selector: '.track-container',
+    });
+    this.tracks = $({ target: this.container, selector: '.tracks' });
+
+    this.resultContainer = $({
+      target: this.container,
+      selector: '.result-container',
+    });
+    this.resultWinner = $({
+      target: this.resultContainer,
+      selector: '.winner',
+    });
+    this.restartButton = $({
+      target: this.resultContainer,
+      selector: '.restart',
+    });
+  }
+
+  bindEvent() {
+    addEvent({
+      el: this.carnameButton,
+      type: 'click',
+      callback: this.registerCars.bind(this),
+    });
+
+    addEvent({
+      el: this.gamecountButton,
+      type: 'click',
+      callback: this.registerCount.bind(this),
+    });
+
+    addEvent({
+      el: this.restartButton,
+      type: 'click',
+      callback: this.restartGame.bind(this),
+    });
+  }
+
+  registerCars = () => {
+    const carNames = this.carnameInput.value.split(',');
+    const validCarNames =
+      carNames &&
+      carNames.every(car => car.trim().length > 0 && car.trim().length <= 5);
+    if (!validCarNames) {
+      alert(MESSAGE.NAME_ERROR);
+      return;
+    }
+
+    this.carnameInput.setAttribute('readOnly', true);
+    this.carnameButton.setAttribute('disabled', true);
+    carNames.map(carname => this.cars.push(new Car(carname)));
+    this.gamecountContainer.classList.remove('d-none');
```

Comment:
push 하는 것도 좋지만, 배열에 대한 불변성을 지키면 더 좋을 것 같아요 😃 
```suggestion
    this.carnameButton.setAttribute('disabled', true);
    carNames.map(carname => this.cars=[...this.cars,new Car(carname)));
    this.gamecountContainer.classList.remove('d-none');
```